src/LayerPlan.cpp

```diff
@@ -2152,6 +2153,238 @@ void LayerPlan::addLinesInGivenOrder(
+    std::optional<TravelAntiOozing>& retraction_amounts,
+    std::optional<TravelAntiOozing>& priming_amounts) const
+{
+    // First compute the actual durations of the travel/z-hop move, as the retraction/prime will have to fit within the travel move without changing this
+    const TravelDurations travel_durations = computeTravelDurations(gcode, extruder, path, path.perform_z_hop ? z_hop_height : 0);
+
+    // Compute the desired retraction distance and duration during travel/z-hop
+    const double retract_distance = retraction_config->retraction_config.distance;
+    const Duration retract_duration = retract_distance / retraction_config->retraction_config.speed;
+    const Ratio retract_during_travel_ratio = gcode.machineHandlesRetraction() ? 0.0_r : retraction_config->retraction_config.retract_during_travel;
+    const Duration retract_duration_during_zhop_and_travel = retract_duration * retract_during_travel_ratio;
+    const Duration retract_duration_during_travel = std::max(0.0_s, retract_duration_during_zhop_and_travel - travel_durations.z_hop);
+
+    // Compute the desired priming distance and duration during travel/z-hop
+    const double prime_distance = retraction_config->retraction_config.distance + gcode.mm3ToE(retraction_config->retraction_config.prime_volume);
+    const Duration prime_duration = prime_distance / retraction_config->retraction_config.primeSpeed;
+    const Ratio prime_during_travel_ratio = gcode.machineHandlesRetraction() ? 0.0_r : retraction_config->retraction_config.prime_during_travel;
+    const Duration prime_duration_during_zhop_and_travel = prime_duration * prime_during_travel_ratio;
+    const Duration prime_duration_during_travel = std::max(0.0_s, prime_duration_during_zhop_and_travel - travel_durations.z_hop);
+
+    // Now check whether we actually have enough time during z-hop + travel to fit the retraction and priming
+    const Duration total_anti_ooze_duration_during_travel = retract_duration_during_travel + prime_duration_during_travel;
+
+    Duration extra_time_still_retraction;
+    Duration extra_time_still_prime;
+    if (total_anti_ooze_duration_during_travel > travel_durations.travel)
+    {
+        // We won't have enough time to perform retraction and priming during travel, so we will have to shift some of it during stationary retract/prime
+        const Ratio retract_prime_ratio = retract_duration_during_travel / total_anti_ooze_duration_during_travel;
+
+        const Duration allowed_retract_duration_during_travel = travel_durations.travel * retract_prime_ratio;
+        extra_time_still_retraction = retract_duration_during_travel - allowed_retract_duration_during_travel;
+
+        const Duration allowed_prime_duration_during_travel = travel_durations.travel * (1.0_r - retract_prime_ratio);
+        extra_time_still_prime = prime_duration_during_travel - allowed_prime_duration_during_travel;
+    }
+
+    const auto compute_anti_ooze_amounts
```

Comment:
If would make this an actual function.This would not only improve readability but also make it easier to write unit tests against the logic for computing the per‑segment amounts.